src/fun_utils/cache.clj

```diff
@@ -8,7 +9,12 @@
 
 (defn ^CacheLoader cache-loader [f]
   (proxy [CacheLoader] []
-    (load [k] (f k))))
+    (load [k] (f k))
+    (reload [k oldvalue]
+            (let [fut (SettableFuture/create)]
+              (future (.set fut (f k)))
```

Comment:
if you use a dynamic binding variable here called *loader-executor* or other, by default it should be clojure.lang.Agent/soloExecutor which is the default future uses, call
(.submit ^ExecutorService *loader-executor* ^Callable #(.set fut (f k))) taking care of relection.

this would allow anyone to provide an alternative thread pool depending on the cache use.